### PR TITLE
feat: add `egctl experimental dashboard  envoy-proxy` to render the admin dashboard for the proxy

### PR DESCRIPTION
**Pull request by hanxiaop**
**What type of PR is this?**

/feature

**What this PR does / why we need it**:

Render the admin dashboard for the proxy, to see things like configdumps without extra steps.

### CODE REVIEW

**Comment by hanxiaop** [comment]
Sample cmd out:
```
 ./egctl x dashboard envoy-proxy -n envoy-gateway-system envoy-engw-eg-a9c23fbb-558f94486c-82wh4
http://localhost:19000
```

File: egctl
Comment (zirain) [inline egctl:1]: please kindly remove this

Comment (hanxiaop) [reply egctl:1]: Thanks, removed.

File: internal/cmd/egctl/envoy_dashboard.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	}
+	dashboardCmd.PersistentFlags().StringArrayVarP(&labelSelectors, "labels", "l", nil, "Labels to select the envoy proxy pod.")
+	dashboardCmd.PersistentFlags().StringVarP(&podNamespace, "namespace", "n", "envoy-gateway-system", "Namespace where envoy proxy pod are installed.")
+	dashboardCmd.PersistentFlags().IntVarP(&listenPort, "port", "p", 0, "Local port to listen to.")
+
+	return dashboardCmd
+}
+
+// portForward first tries to forward localhost:remotePort to podName:remotePort, falls back to dynamic local port
+func portForward(podName, namespace, urlFormat string, listenPort int, client kube.CLIClient, writer io.Writer) error {
```
[the remark this answers is not in the document]
Comment (hanxiaop) [reply internal/cmd/egctl/envoy_dashboard.go:80]: This uses the port-forwarder logic, as you can see in https://github.com/envoyproxy/gateway/pull/2669/files#diff-926d891150379e2197ee1682b6296723490999ab9fe9623a32f00e79337d327cR86. It's not implementing new logics but wrapping with additional logics to open the browser from the terminal.

[the remark this answers is not in the document]
Comment (shawnh2) [reply internal/cmd/egctl/envoy_dashboard.go:80]: nvm 🥲 missing that, thanks

File: internal/cmd/egctl/envoy_dashboard.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	kube "github.com/envoyproxy/gateway/internal/kubernetes"
+)
+
+func newEnvoyDashboardCmd() *cobra.Command {
+	var podName, podNamespace string
+	var listenPort int
+
+	dashboardCmd := &cobra.Command{
+		Use:   "envoy-proxy <name> -n <namespace>",
+		Short: "Retrieves Envoy admin dashboard for the specified pod",
```
Comment (shawnh2) [inline internal/cmd/egctl/envoy_dashboard.go:29]:
```suggestion
		Short: "Retrieve Envoy admin dashboard for the specified pod",
```

File: internal/cmd/egctl/envoy_dashboard.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+// portForward first tries to forward localhost:remotePort to podName:remotePort, falls back to dynamic local port
+func portForward(podName, namespace, urlFormat string, listenPort int, client kube.CLIClient, writer io.Writer) error {
+	var fw kube.PortForwarder
+	meta := types.NamespacedName{
+		Namespace: namespace,
+		Name:      podName,
+	}
+	fw, err := kube.NewLocalPortForwarder(client, meta, listenPort, adminPort)
```
Comment (shawnh2) [inline internal/cmd/egctl/envoy_dashboard.go:86]: port number should be checked, whether it is in a valid range

File: internal/cmd/egctl/envoy_dashboard.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+	dashboardCmd := &cobra.Command{
+		Use:   "envoy-proxy <name> -n <namespace>",
+		Short: "Retrieves Envoy admin dashboard for the specified pod",
+		Long:  `Retrieve Envoy admin dashboard for the specified pod.`,
+		Example: `  # Retrieve Envoy admin dashboard for the specified pod.
+  egctl experimental dashboard envoy-proxy <pod-name> -n <namespace>
+
+  # short syntax
+  egctl experimental d envoy-proxy <pod-name> -n <namespace>
```
Comment (shawnh2) [inline internal/cmd/egctl/envoy_dashboard.go:35]: can you add a section in https://github.com/envoyproxy/gateway/blob/main/site/content/en/latest/user/egctl.md to describe more about this subcommand?

**Comment by hanxiaop** [comment]
@shawnh2 Thank you for the review! I think I have addressed all the comments.

**Review by shawnh2 — APPROVED** [review]
LGTM, thanks

**Review by zirain — APPROVED** [review]

**Pull request merged**